--- src/app/code/core/Mage/Index/Model/Process.php	2014-05-14 16:03:36.000000000 +0200
+++ src/app/code/core/Mage/Index/Model/Process_new.php	2014-10-24 23:38:29.000000000 +0200
@@ -475,16 +475,9 @@
      */
     public function isLocked()
     {
-        if ($this->_isLocked !== null) {
-            return $this->_isLocked;
-        } else {
-            $fp = $this->_getLockFile();
-            if (flock($fp, LOCK_EX | LOCK_NB)) {
-                flock($fp, LOCK_UN);
-                return false;
-            }
-            return true;
-        }
+        // flock() with LOCK_EX is not working on NFS filesystem when using Vagrant, and it will always fail
+        // We disable the lock mechanism by returning every time 'false'
+        return false;
     }
 
     /**
